### PR TITLE
DXCDT-439: Add new `auth0_organization_member_role` resource

### PR DESCRIPTION
**Pull request by sergiught**
### 🔧 Changes

Add new resource to manage the 1:1 relationship between an organization member and it's roles. 

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: internal/auth0/organization/resource_member_role_test.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+					resource.TestCheckResourceAttr("auth0_organization_member_role.role1", "role_name", fmt.Sprintf("Test Reader - %s", testName)),
+					resource.TestCheckResourceAttr("auth0_organization_member_role.role1", "role_description", ""),
+					resource.TestCheckResourceAttrSet("auth0_organization_member_role.role2", "user_id"),
+					resource.TestCheckResourceAttrSet("auth0_organization_member_role.role2", "organization_id"),
+					resource.TestCheckResourceAttr("auth0_organization_member_role.role2", "role_name", fmt.Sprintf("Test Writer - %s", testName)),
+					resource.TestCheckResourceAttr("auth0_organization_member_role.role2", "role_description", ""),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccGivenTwoRolesAUserAnOrganizationAndAnOrganizationMember, testName),
```
Comment (willvedd) [inline internal/auth0/organization/resource_member_role_test.go:82]: Isn't this what removes the member's roles?

File: internal/auth0/organization/resource_member_role_test.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+				Config: acctest.ParseTestName(testAccRemoveAssignedRolesFromOrganizationMember, testName),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectEmptyPlan(),
+					},
+				},
+			},
+		},
+	})
+}
```
Comment (willvedd) [inline internal/auth0/organization/resource_member_role_test.go:113]: I _think_ I get the gist of it? Perform a delete by removing the 1:1 and then tracking diff changes with the 1:many and asserting that there are no roles existing. Am I correct in my understanding?

Comment (willvedd) [inline internal/auth0/organization/resource_member_role_test.go:113]: Missing a test for the `auth0_organization_member_role` import itself. Only importing the 1:many above.

File: internal/auth0/organization/resource_member_role.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+// NewMemberRoleResource will return a new auth0_organization_member_role (1:1) resource.
+func NewMemberRoleResource() *schema.Resource {
+	return &schema.Resource{
+		Description:   "This resource is used to manage the roles assigned to an organization member.",
+		CreateContext: createOrganizationMemberRole,
+		ReadContext:   readOrganizationMemberRole,
+		DeleteContext: deleteOrganizationMemberRole,
+		Importer: &schema.ResourceImporter{
+			StateContext: internalSchema.ImportResourcePairID("organization_id", "user_id"),
```
Comment (willvedd) [inline internal/auth0/organization/resource_member_role.go:24]: Appears that the role ID is missing from the import process. Probably requires modification to the helper function or require bespoke logic.

File: internal/auth0/organization/resource_member_role.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+	if err := api.Organization.AssignMemberRoles(organizationID, userID, []string{roleID}); err != nil {
+		if err, ok := err.(management.Error); ok && err.Status() == http.StatusNotFound {
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	data.SetId(organizationID + ":" + userID)
```
Comment (willvedd) [inline internal/auth0/organization/resource_member_role.go:78]: Need to attach the role ID to this too to enable management of multiple roles for a org member.

File: internal/acctest/resource.go [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+		return "", fmt.Errorf("extract resource attribute: failed to find resource with name: %q", name)
+	}
+
+	attribute, ok := tfResource.Primary.Attributes[key]
+	if !ok {
+		return "", fmt.Errorf("extract resource attribute: failed to find attribute with name: %q", key)
+	}
+
+	return attribute, nil
+}
```
Comment (willvedd) [inline internal/acctest/resource.go:22]: I know we could've taken this in a number of directions but this implementation seems like the best.

**Review by willvedd — CHANGES_REQUESTED** [review]
If I understand correctly, looks like the import function may need adjustment but otherwise it looks good.

File: internal/auth0/organization/resource_member_role_test.go (hunk 2 above)
Comment (sergiught) [reply internal/auth0/organization/resource_member_role_test.go]: Yes.

Comment (sergiught) [reply internal/auth0/organization/resource_member_role_test.go]: Updated!

File: internal/auth0/organization/resource_member_role.go (hunk 3 above)
Comment (sergiught) [reply internal/auth0/organization/resource_member_role.go]: Updated!

**Review by willvedd — APPROVED** [review]

**Pull request merged**